include/zephyr/net/conn_mgr_monitor.h

```diff
@@ -95,9 +148,98 @@ void conn_mgr_watch_l2(const struct net_l2 *l2);
 #define conn_mgr_watch_iface(...)
 #define conn_mgr_ignore_l2(...)
 #define conn_mgr_watch_l2(...)
+#define conn_mgr_register_online_checker_cb(...)
 
 #endif /* CONFIG_NET_CONNECTION_MANAGER */
 
+/** @cond INTERNAL_HIDDEN */
+#if defined(CONFIG_NET_CONNECTION_MANAGER_ONLINE_CONNECTIVITY_CHECK)
+/* Let online checker to decide if this network packet can be considered
+ * when deciding if we are online or not.
+ */
+void conn_mgr_online_checker_update(struct net_pkt *pkt,
+				    void *hdr,
+				    sa_family_t family,
+				    bool is_loopback);
+
+/* Used when triggering online connectivity check manually from net-shell
+ * and "net conn check" command. This is mainly for debugging purposes.
+ */
+bool conn_mgr_trigger_online_connectivity_check(void);
+#else
+static inline void conn_mgr_online_checker_update(struct net_pkt *pkt,
+						  void *hdr,
+						  sa_family_t family,
+						  bool is_loopback)
+{
+	ARG_UNUSED(pkt);
+	ARG_UNUSED(hdr);
+	ARG_UNUSED(family);
+	ARG_UNUSED(is_loopback);
+}
+
+static inline bool conn_mgr_trigger_online_connectivity_check(void)
+{
+	return false;
+}
+
+#endif
+/** @endcond */
+
+/** Possible choices for the online connectivity check. */
+enum net_conn_mgr_online_check_type {
+	/** Use ICMP Echo-Request (ping). This is the default */
+	NET_CONN_MGR_ONLINE_CHECK_PING = 0,
+	/** Use HTTP(S) GET request. */
+	NET_CONN_MGR_ONLINE_CHECK_HTTP,
+};
```

Comment:
I actually had this as Kconfig configured only at the beginning, so user was only able to either select ping or http but not both. But then realized that it actually makes it slightly cumbersome to use so made it so that user can select both if needed and made it runtime configurable. This runtime configurability can useful for example depending on application needs / networks etc.

Reply:
But I am not convinced there is a use-case where the application actually needs to switch back and forth, and having all the settings for connectivity check be run-time configurable significantly increases the API complexity, so I think it is worth it to simplify the API and remove the unnecessary runtime customization

I just cannot think of a reason an application would want to change this from whatever was configured at build time